internal/store/fleetdb/fleetdb.go

```diff
@@ -387,6 +388,23 @@ func (r *Store) createUpdateServerComponents(ctx context.Context, serverID uuid.
 
 		_, err = r.CreateComponents(ctx, serverID, add)
 		if err != nil {
+			// Log the specific component if it causes duplicate key error.
+			if strings.Contains(err.Error(), "duplicate key value violates unique constraint") {
+				pattern := `serial, server_component_type_id\)=\('.*', '([^']*)', '.*'\)`
```

Comment:
Lets just log the contents of `add` as is, so its something like ..
```
r.logger.WithFields(logrus.Fields{"create-components": add, "err": err.Error()}).WithError("create components returned duplicate key error")
```

What do you think?

Reply:
SGTM.

I was worried about dumping too much logs will make it difficult to search for what we are looking for. But we could use `ctrl + f` lmao :p

Do you think it may log sensitive information?

Reply:
That is a valid concern. Since we don't have access to the component data that resulted in the error, I don't see another way. We have alerting for Alloy failures that should notify us and so we can respond to fixing the situation quicker with this data. 

There is no sensitive information within the `add`, `update` payloads.